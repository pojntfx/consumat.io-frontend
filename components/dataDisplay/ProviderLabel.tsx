import { Provider } from "../../lib/api/consumat-io";

type ProviderLabelProps = {
  provider: Provider;
};

function ProviderLabel({ provider }: ProviderLabelProps) {
  return (
    <div className="flex items-center justify-center mr-1 py-1 px-2 bg-gray-200 dark:bg-gray-800 text-white text-sm font-bold rounded shadow">
      {provider.name === "Netflix" ? (
        <img src="/images/netflix.png" className="h-6" />
      ) : provider.name === "Amazon Prime Video" ? (
        <img src="/images/prime-video.png" className="h-6" />
      ) : provider.name === "Disney Plus" ? (
        <img src="/images/disney-plus.png" className="h-6" />
      ) : (
        <div>{provider.name}</div>
      )}
    </div>
  );
}

export default ProviderLabel;
