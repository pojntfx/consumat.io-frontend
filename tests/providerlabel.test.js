import React from "react";
import renderer from "react-test-renderer";
import ProviderLabel from "../components/dataDisplay/ProviderLabel";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(<ProviderLabel provider={testExamples.exampleProvider} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
