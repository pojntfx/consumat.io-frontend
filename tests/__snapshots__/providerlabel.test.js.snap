// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex items-center justify-center mr-1 py-1 px-2 bg-gray-200 dark:bg-gray-800 text-white text-sm font-bold rounded shadow"
>
  <div />
</div>
`;
